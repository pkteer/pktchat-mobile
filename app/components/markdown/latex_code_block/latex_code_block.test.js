// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {shallow} from 'enzyme';
import React from 'react';

import Preferences from '@mm-redux/constants/preferences';

import LatexCodeBlock from './latex_code_block';

describe('LatexCodeBlock', () => {
    const baseProps = {
        theme: Preferences.THEMES.denim,
        language: 'latex',
        textStyle: {},
    };

    test('should match snapshot', () => {
        const wrapper = shallow(
            <LatexCodeBlock
                {...baseProps}
                content={'\\frac{1}{2} == 0.5^{0*\\pi} - 0.5'}
            />,
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should have 2 lines', () => {
        const wrapper = shallow(
            <LatexCodeBlock
                {...baseProps}
                content={'\\frac{2}{1} = 2 \\\\ 2_{nd} = line'}
            />,
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should have moreLinesText', () => {
        const wrapper = shallow(
            <LatexCodeBlock
                {...baseProps}
                content={'\\frac{5}{2} = 2 \\\\ 2_{nd} = line \\\\ \\pi = 3.14'}
            />,
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });
});
