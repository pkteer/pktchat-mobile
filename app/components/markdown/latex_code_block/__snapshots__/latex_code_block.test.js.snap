// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LatexCodeBlock should have 2 lines 1`] = `
<TouchableWithFeedbackIOS
  onLongPress={[Function]}
  onPress={[Function]}
  type="opacity"
>
  <View
    style={
      Object {
        "borderColor": "rgba(63,67,80,0.15)",
        "borderRadius": 3,
        "borderWidth": 0.5,
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "paddingLeft": 6,
          "paddingVertical": 4,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "marginLeft": 5,
            "paddingVertical": 4,
          }
        }
      >
        <ForwardRef
          color="black"
          math="\\\\frac{2}{1} = 2"
          onError={[Function]}
          renderError={[Function]}
          resizeMode="cover"
        />
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "marginLeft": 5,
            "paddingVertical": 4,
          }
        }
      >
        <ForwardRef
          color="black"
          math="2_{nd} = line"
          onError={[Function]}
          renderError={[Function]}
          resizeMode="cover"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#192a4d",
          "justifyContent": "center",
          "opacity": 0.8,
          "padding": 6,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ffffff",
            "fontSize": 12,
          }
        }
      >
        LaTeX
      </Text>
    </View>
  </View>
</TouchableWithFeedbackIOS>
`;

exports[`LatexCodeBlock should have moreLinesText 1`] = `
<TouchableWithFeedbackIOS
  onLongPress={[Function]}
  onPress={[Function]}
  type="opacity"
>
  <View
    style={
      Object {
        "borderColor": "rgba(63,67,80,0.15)",
        "borderRadius": 3,
        "borderWidth": 0.5,
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "paddingLeft": 6,
          "paddingVertical": 4,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "marginLeft": 5,
            "paddingVertical": 4,
          }
        }
      >
        <ForwardRef
          color="black"
          math="\\\\frac{5}{2} = 2"
          onError={[Function]}
          renderError={[Function]}
          resizeMode="cover"
        />
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "marginLeft": 5,
            "paddingVertical": 4,
          }
        }
      >
        <ForwardRef
          color="black"
          math="2_{nd} = line"
          onError={[Function]}
          renderError={[Function]}
          resizeMode="cover"
        />
      </View>
      <InjectIntl(FormattedText)
        defaultMessage="+{count, number} more {count, plural, one {line} other {lines}}"
        id="mobile.markdown.code.plusMoreLines"
        style={
          Object {
            "color": "rgba(63,67,80,0.4)",
            "fontSize": 11,
            "marginTop": 2,
          }
        }
        values={
          Object {
            "count": 1,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#192a4d",
          "justifyContent": "center",
          "opacity": 0.8,
          "padding": 6,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ffffff",
            "fontSize": 12,
          }
        }
      >
        LaTeX
      </Text>
    </View>
  </View>
</TouchableWithFeedbackIOS>
`;

exports[`LatexCodeBlock should match snapshot 1`] = `
<TouchableWithFeedbackIOS
  onLongPress={[Function]}
  onPress={[Function]}
  type="opacity"
>
  <View
    style={
      Object {
        "borderColor": "rgba(63,67,80,0.15)",
        "borderRadius": 3,
        "borderWidth": 0.5,
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "paddingLeft": 6,
          "paddingVertical": 4,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "marginLeft": 5,
            "paddingVertical": 4,
          }
        }
      >
        <ForwardRef
          color="black"
          math="\\\\frac{1}{2} == 0.5^{0*\\\\pi} - 0.5"
          onError={[Function]}
          renderError={[Function]}
          resizeMode="cover"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#192a4d",
          "justifyContent": "center",
          "opacity": 0.8,
          "padding": 6,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ffffff",
            "fontSize": 12,
          }
        }
      >
        LaTeX
      </Text>
    </View>
  </View>
</TouchableWithFeedbackIOS>
`;
